Intersect.Server/Entities/Events/Conditions.cs

```diff
@@ -390,9 +390,16 @@ QuestBase questBase
                 var entities = mapInstance.GetEntities();
                 foreach (var en in entities)
                 {
-                    if (en.GetType() == typeof(Npc))
+                    if (en is Npc npc)
                     {
-                        return false;
+                        if (!condition.SpecificNpc)
+                        {
+                            return false;
+                        }
+                        else if (npc.Base?.Id == condition.NpcId)
+                        {
+                            return false;
+                        }
```

Comment:
uhm, can this be combined with an `||` 😂

Reply:
LOL, thank you. I knew something was dumb about this when I was writing it. Done.